### PR TITLE
October Program Updates

### LINKED ISSUE #77: Front-end: update October Events
# Purpose
[October Events](https://github.com/Code-4-Community/outstanding-life/pull/71) was merged in because we were on a tight schedule, but there are still some PR comments that need to be addressed. David also sent back some edits, so that's included in this as well. 

# Description

**Questions to answer:**
* Are we able to get the header divider in the Programs page span the whole page (rather than just fit under "October 2022 Online Events")?
* [PR](https://github.com/Code-4-Community/outstanding-life/pull/71#discussion_r994843044): Why are we adding the whiteSpace="pre-line" attribute? This replaces white space with more than 2 spaces with a single space. Is our content just not formatted correctly?
* [PR](https://github.com/Code-4-Community/outstanding-life/pull/71#discussion_r994919979): Is this really the best way to store this information? At this point a line by line string is probably a bad representation.
* [PR](https://github.com/Code-4-Community/outstanding-life/pull/71#discussion_r994920208): Did you type this all in manually?

**Tasks**
- [ ] Monthly Writing Group: please replace the description with this:
> In this monthly writing workshop, we will discuss a short excerpt from fiction, non-fiction, or poetry. We will discuss its effectiveness either in form, style, point of view, or other literary devices/techniques. After considering what makes the passage work, we will use some aspect of the text as a prompt to jumpstart our own writing. The focus will be on short, in-class writing and discussion. Some experience in writing is required.

Referencing the screenshot below: 
- [ ] Please insert a new line between the two bios to make it more readable
- [ ] Would it be possible to either underline, italicize, or bold the "Bios of workshop leaders" line? Right now it kinda looks like it's just floating there
Screen Shot 2022-10-19 at 9 40 45 PM

- [ ] Following Izzy's [designs](https://www.figma.com/file/uKPcL5G1ZwVC3SGuacJco6/Outstanding-Life?node-id=1615%3A10586), implement a dropdown that allows you to switch between Upcoming Events and Past Events
Screen Shot 2022-10-19 at 9 47 18 PM

# Acceptance Criteria
- [ ] Please attach before and after screenshots of the changes you've made to these pages.

### PR DESCRIPTION
**Pull request by al-chang**
### ℹ️ Issue

Closes #77

### ✔️ Verification
Screen Shot 2022-11-06 at 7 03 41 PM
Screen Shot 2022-11-06 at 7 03 57 PM
Screen Shot 2022-11-06 at 7 04 10 PM

### 🏕️ (Optional) Future Work / Notes

N/A

### CODE REVIEW

File: pages/programs/upcoming-programs/index.tsx [hunk 1]
```diff
@@ -1,18 +1,17 @@
-import { ProgramListingProps, ProgramsList } from '../../lib/components/programs-list';
+import { ProgramListingProps, ProgramsList } from '../../../lib/components/programs-list';
 import { Container, Flex } from '@chakra-ui/react';
-import OSLHeader from '../../lib/components/header';
+import OSLHeader from '../../../lib/components/header';
 
 const programs: ProgramListingProps[] = [
```
Comment (adriannaliu) [inline pages/programs/upcoming-programs/index.tsx:5]: nit: Can we call this upcomingPrograms?

**Review by adriannaliu — COMMENTED** [review]
lgtm so far! only did a quick skim so just one very small nit rn, but i'll come back to this later

File: lib/components/navbar/NavLink.tsx [hunk 2]
```diff
@@ -1,30 +1,52 @@
...
+        <Box
+          border="3px solid var(--magenta)"
+          position="absolute"
+          width="100%"
+          background="#FFF"
+          zIndex="100">
+          {dropDownOptions.map(({ label, path }) => (
+            <Box
+              key={label}
+              backgroundColor={router.pathname === path ? '#c2c4c8' : undefined}
```
Comment (thejamielin) [inline lib/components/navbar/NavLink.tsx:40]: Could we take this color out into a global variable so this part is a bit more readable? Maybe gray?

Comment (thejamielin) [inline lib/components/navbar/NavLink.tsx:40]: We could the isActive variable from above here as well so we don't have to repeat the equality check

File: lib/components/navbar/NavLink.tsx [hunk 3]
```diff
@@ -1,30 +1,52 @@
...
-        </chakra.span>
-      </ChakraLink>
-    </Link>
+    <Box position="relative" onMouseEnter={onOpen} onMouseLeave={onClose}>
+      <Link href={path} passHref>
+        <ChakraLink
+          _hover={{ cursor: 'pointer' }}
+          height="fit-content"
+          data-cy={'nav-bar-' + label}
+          color="#58595B"
```
Comment (thejamielin) [inline lib/components/navbar/NavLink.tsx:18]: Could we pull out this color into a global variable to make this more readable?

File: lib/components/navbar/NavLink.tsx [hunk 4]
```diff
@@ -1,30 +1,52 @@
...
+          textDecorationThickness="3px"
+          textDecorationColor={'var(--magenta)'}
+          tabIndex={0}
+          _focus={{ boxShadow: 'none' }}>
+          <chakra.span fontWeight="bold" fontSize="24px">
+            {label}
+          </chakra.span>
+        </ChakraLink>
+      </Link>
+      {dropDownOptions && isOpen && (
```
Comment (thejamielin) [inline lib/components/navbar/NavLink.tsx:30]: this is literally amazing. great work @al-chang and @nithyabalaji !

File: lib/components/navbar/NavLink.tsx [hunk 5]
```diff
@@ -1,30 +1,52 @@
...
+        <Box
+          border="3px solid var(--magenta)"
+          position="absolute"
+          width="100%"
+          background="#FFF"
```
Comment (thejamielin) [inline lib/components/navbar/NavLink.tsx:35]: Same comment on color, what color is this?

**Review by thejamielin — CHANGES_REQUESTED** [review]
@al-chang and @nithyabalaji great job guys! left a couple of comments but should be quick fixes. Amazing work and we can merge this very soon!!!!!

File: pages/programs/programs.json [hunk 6]
```diff
@@ -0,0 +1,16 @@
+[
```
[the remark this answers is not in the document]
Comment (al-chang) [reply pages/programs/programs.json:1]: Where could we find a list of November events?

[the remark this answers is not in the document]
Comment (adriannaliu) [reply pages/programs/programs.json:1]: @al-chang if you merge in master the November events should be there!

**Review by adriannaliu — COMMENTED** [review]
lgtm guys, please take a look at jamie's comments!

**Review by jessicasu77 — DISMISSED** [review]
LGTM! agreed w/ the comments that Jamie & Adri made

File: lib/components/navbar/NavLink.tsx (hunk 2 above)
Comment (al-chang) [reply lib/components/navbar/NavLink.tsx]: This is using the path for the dropdown option which comes from the props, that variable is checking for the route of the menu bar item and is a different value. I tried to play around with a cleaner way of doing this but couldn't get anything better without breaking it out into its own component which I would argue is unnecessary given the tight coupling of these items.

**Review by thejamielin — DISMISSED** [review]
Looks about ready to merge! Is there a reason we haven't abstracted PastPrograms and UpcomingPrograms?

**Review by thejamielin — APPROVED** [review]

**Review by adriannaliu — APPROVED** [review]
looks great!

**Pull request merged**